spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/provisioning/KinesisStreamProvisioner.java

```diff
@@ -139,6 +139,9 @@ public ConsumerDestination provisionConsumerDestination(String name, String grou
 				}
 			}
 			catch (ResourceNotFoundException ex) {
+				if (!this.configurationProperties.isAutoCreateStream()) {
+					throw ex;
```

Comment:
Please, wrap this into the `ProvisioningException` with some clear message why we have failed by the `autoCreateStream = false` reason.